draft-ietf-quic-transport.md

```diff
@@ -5277,15 +5277,16 @@ Largest Acknowledged:
 
 ACK Delay:
 
-: A variable-length integer representing the time delta in microseconds between
-  when the ACK frame was sent and when the largest acknowledged packet, as
-  indicated in the Largest Acknowledged field, was received by this peer.  The
-  value of the ACK Delay field is scaled by multiplying the encoded value by 2
-  to the power of the value of the ack_delay_exponent transport parameter set by
-  the sender of the ACK frame; see {{transport-parameter-definitions}}.  Scaling
-  in this fashion allows for a larger range of values with a shorter encoding at
-  the cost of lower resolution.  Because the receiver doesn't use the ACK Delay
-  for Initial and Handshake packets, a sender SHOULD send a value of 0.
+: A variable-length integer representing the intentional delay in microseconds
+  between when the largest acknowledged packet, as indicated in the Largest
+  Acknowledged field, was received, and when the ACK frame was sent;
+  see {{host-delay}}.  The value of the ACK Delay field is scaled by
```

Comment:
Including the encoding/unit details in the description makes this a very hard sentence to swallow.  I can only think of ways to make this longer though...

This value represents the length of the delay between when the largest acknowledged packet, as indicated in the Largest Acknowledged field, was received, and when the ACK frame was sent; see {{host-delay}}.  ACK Delay is encoded as a variable-length integer with a value that is derived by dividing the delay in microseconds by 2 to the power of the value of the ack_delay_exponent transport parameter set by the sender of the ACK frame; see {{transport-parameter-definitions}}.  To decode the value, the encoded variable-length integer value is multiplied by 2 to the power of the ack_delay_exponent transport parameter to obtain a value in microseconds.  Scaling in this fashion allows ...

Reply:
I flipped the ordering of this sentence, but I agree it was already too long.  I'm going to work off of @martinthomson suggestion and move the units to a separate sentence.

Reply:
I tried to reword, thought in the end I'm not sure I ended up using @martinthomson text or not.

Removing microseconds from the first sentence makes this clearer to me, since the ACK Delay field is definitely not in microseconds, the decoded acknowledgement delay is.

Reply:
@ianswett: You're going to conflict with changes in #3946, which rephrases this as well.

Reply:
Agreed, I can fix the conflicts if this lands second.  I'm more concerned with getting this text correct.

Reply:
Ok, I've done a full rework of the paragraph.